tests/manage/pv_services/test_delete_project_with_pvc_and_pods.py

```diff
@@ -0,0 +1,175 @@
+"""
+The purpose of this test case is to delete at least 2 projects:
+One project with just pvcs and the other with pvcs attached to a pod
+"""
+import logging
+import time
+import pytest
+from itertools import cycle
+
+from ocs_ci.framework.testlib import ManageTest, tier1
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.resources.ocs import OCS
+from tests import helpers
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@pytest.mark.polarion_id("OCS-278")
+@pytest.mark.parametrize(
+    argnames=["pvcs_num"],
+    argvalues=[
+        pytest.param(
+            100, marks=pytest.mark.tier1
+        ),
+        pytest.param(
+            1000, marks=pytest.mark.scale
+        )
+    ]
+)
+class TestDeleteProjectWithPVCAndPods(ManageTest):
+    """
+    Create at least 2 projects, one with just pvcs and the other with pvcs
+    attached to pod, and delete them.
+    """
+
+    @pytest.fixture()
+    def setup(self, storageclass_factory):
+        """
+        Create an OCP+OCS Setup, as well as two Storage classes:
+        one for CephFS and one for RBD.
+        Scale the setup to at-least 800 pvcs/pods.
+        """
+        self.ocp_setup = ocp.OCP()
+        self.ocs_setup = OCS()  # TODO: Scale the setup to at-least 800 pvcs/pods
+        self.ceph_cluster = CephCluster()
+
+        log.info("Creating CephFS Storage class")
+        self.cephfs_sc = storageclass_factory(interface=constants.CEPHFILESYSTEM)
+        log.info("Creating RBD Storage class")
+        self.rbd_sc = storageclass_factory(interface=constants.CEPHBLOCKPOOL)
+
+    def test_delete_project_with_pvc_and_pods(
+        self, setup, project_factory, multi_pvc_factory,
+        pvc_factory, pod_factory, pvcs_num
+    ):
+        # Start with Project 1
+        log.info("Creating Project")
+        project_1 = project_factory()
+
+        log.info(f"Creating {pvcs_num} CephFS PVCs")
+        # Generate a given number of CephFS PVCs, some RWO and some RWX
+        rwo_rwx = cycle([constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX])
+        for i in range(0, pvcs_num):
+            pvc_factory(interface=constants.CEPHFILESYSTEM, project=project_1,
+                        storageclass=self.cephfs_sc, access_mode=next(rwo_rwx))
+        log.info(f"Creating {pvcs_num} RBD PVCs")
+        multi_pvc_factory(interface=constants.CEPHBLOCKPOOL,
+                          project=project_1,
+                          storageclass=self.rbd_sc,
+                          access_mode=constants.ACCESS_MODE_RWO,
+                          num_of_pvc=pvcs_num)
+
+        # Delete the entire Project 1 (along with all of its PVCs)
+        pvs = helpers.get_all_pvs()
```

Comment:
Isn't this giving the list of all PVs in the cluster ?

Reply:
@hadarsharon this seems like risky